code/obj/item/device/pda2/smallprogs.dm

```diff
@@ -514,79 +514,106 @@ Code:
 	var/obj/machinery/power/pt_laser/laser
 	var/obj/machinery/power/generatorTemp/generator
 	var/list/obj/machinery/power/collector_control/collector_controlers
+	var/list/obj/machinery/power/vent_capture/vents
+	var/obj/machinery/carouselpower/carousel
+
+	New()
+		..()
+		collector_controlers = list()
+		vents = list()
+
+	proc/find_machinery(obj/ref, type)
+		if(!ref || ref.disposed)
+			ref = locate(type) in machine_registry[MACHINES_POWER]
+			if(ref?.z != 1) ref = null
+		. = ref
+
+	proc/find_machinery_list(list/obj_list, type)
+		obj_list.len = 0
+		for(var/obj/M in machine_registry[MACHINES_POWER])
+			if(istype(M, type) && M.z == 1)
+				obj_list.Add(M)
```

Comment:
..................why are you not just using by_type

Reply:
Excuse: O(n) vs O(n)
Excuse: Space savings, I fall back mentally to embedded space constraints.
Excuse: List of things currently using those lists seems so small, so adding objects that there are typically 0 to 5 of seemed odd.

Will update, good idea. 👍

Reply:
not O(n) vs O(N) btw, O(1) vs O(N)

Reply:
🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦
I thought you were just talking about line 533, and I was thinking I still wanted to filter on z level.  My bad.  When you're right, you're right. 😉 

No excuses.